packages/gatsby/cache-dir/dev-loader.js

```diff
@@ -2,9 +2,53 @@ import { BaseLoader, PageResourceStatus } from "./loader"
 import { findPath } from "./find-path"
 
 class DevLoader extends BaseLoader {
-  constructor(syncRequires, matchPaths) {
-    const loadComponent = chunkName =>
-      Promise.resolve(syncRequires.components[chunkName])
+  constructor(lazySyncRequires, matchPaths) {
+    const loadComponent = (chunkName, path) => {
+      const realPath = findPath(path)
+      delete require.cache[
+        require.resolve(`$virtual/lazy-client-sync-requires`)
+      ]
+      const lazyRequires = require(`$virtual/lazy-client-sync-requires`)
+      console.log(`loadComponent`, { chunkName, lazyRequires })
+      if (lazyRequires.lazyComponents[chunkName]) {
+        return Promise.resolve(lazyRequires.lazyComponents[chunkName])
+      } else {
+        console.log(`trigger compilation on the server for`, chunkName)
+        return new Promise(resolve => {
```

Comment:
This promise doesn't reject (errors would be lost) and the timeout timer does not call reject nor resolve, meaning I think it ends up in a zombie state once it hits that path, since the interval that would call `resolve` and is the only way to resolve the promise is stopped.

Reply:
There's nothing here to reject on I don't think. We're just telling the server to start compiling the page component and then... waiting. We're not doing anything per se.

Reply:
Anything can crash. `this.loadComponent` can fail for any number of reasons :shrug: